test/dual_n.jl

```diff
@@ -86,3 +86,6 @@ sq = squareroot(Dual4(10000.0,1.0,2.0,3.0,4.0))
 @test_approx_eq epsilon2(NaNMath.pow(exp(1),Dual4(1.0,1.0,2.0,3.0,4.0))) 2exp(1)
 @test_approx_eq epsilon3(NaNMath.pow(exp(1),Dual4(1.0,1.0,2.0,3.0,4.0))) 3exp(1)
 @test_approx_eq epsilon4(NaNMath.pow(exp(1),Dual4(1.0,1.0,2.0,3.0,4.0))) 4exp(1)
+
+@test real(mod(dual(15.23, 1), 10)) == 5.23
```

Comment:
The tests are not quite well organized, but this should go into the ``automatic_differentiation_test.jl`` file.